lib/compat.php

```diff
@@ -24,18 +24,90 @@
 	 * @return WP_Block_Type|false The registered block type on success, or false on failure.
 	 */
 	function register_block_type_from_metadata( $path, $args = array() ) {
-		$file = trailingslashit( $path ) . 'block.json';
-		if ( ! file_exists( $file ) ) {
+		$metadata_file = trailingslashit( $path ) . 'block.json';
+		if ( ! file_exists( $metadata_file ) ) {
 			return false;
 		}
 
-		$metadata = json_decode( file_get_contents( $file ), true );
-		if ( ! is_array( $metadata ) ) {
+		$metadata = json_decode( file_get_contents( $metadata_file ), true );
+		if ( ! is_array( $metadata ) || ! $metadata['name'] ) {
```

Comment:
Is this meant to be checking the presence of a `name` property? This syntax will log a notice error.

```
⇒ php -a
Interactive shell

php > error_reporting( E_ALL );
php > $metadata = array();
php > var_export( ! $metadata['name'] );

Notice: Undefined index: name in php shell code on line 1
true
```

```suggestion
		if ( ! is_array( $metadata ) || ! isset( $metadata['name'] ) ) {
```

Or:

```suggestion
		if ( ! is_array( $metadata ) || empty( $metadata['name'] ) ) {
```